samples/gpu/hog.cpp

```diff
@@ -115,11 +115,14 @@ int main(int argc, char** argv)
 {
     try
     {
-        if (argc < 2)
+        Args args;
+        if (argc < 2) {
             printHelp();
-        Args args = Args::read(argc, argv);
-        if (help_showed)
-            return -1;
+            args.camera_id = 0;
+            args.src_is_camera = true;
+        }
+        else
+            args = Args::read(argc, argv);
```

Comment:
In this case, program launched with `--help` option will not exit correctly. Can you confirm this? 
I think following check should be added inside else branch, after this line:
```
if (help_showed) 
    return -1;
```